### ISSUE TITLE
make failed on Debian 9 (stretch)

### ISSUE THREAD

**Issue by prudnitskiy** [emercoin/emercoin#58]
Hi there!

I'm trying to compile an Emercoin from source on Debian 9 (x64), but make failed with error:

```
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
Makefile:3158: recipe for target 'libbitcoin_common_a-chainparams.o' failed
make[2]: *** [libbitcoin_common_a-chainparams.o] Error 1
make[2]: Leaving directory '/var/tmp/emercoin-0.6.3.1emc/src'
Makefile:6423: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/emercoin-0.6.3.1emc/src'
Makefile:578: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

Steps to reproduce:
- install clean Debian 9
- install packages necessary to build: `apt-get install -y build-essential libboost-all-dev autoconf pkg-config autotools-dev libminiupnpc-dev libevent-dev libssl1.0-dev`
- install db4 packages from your repo
- run build as it displayed on documentation:
```
./autogen.sh
./configure --prefix=/opt/emercoin-0.6.3.1
make
```

**Comment by FndNur1Labs** [comment]
how to fixed this 

AC_CONFIG_MACRO_DIRS([build-aux/m4]) conflicts with ACLOCAL.AMFLAGS=-I build-aux/m4

**Comment by EvgenijM86** [comment]
You might also want to read this issue: https://github.com/emercoin/emercoin/issues/63
Here is a copy of a solution that semi-works:

This will build emercoind, emercoin-tx, emercoin-cli, but it will fail for emercoin-qt.

```
apt install libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev
apt install git
apt install autoconf libtool pkg-config
apt install libqt5gui5 libqt5core5a libqt5dbus5 qttools5-dev qttools5-dev-tools libprotobuf-dev protobuf-compiler
apt install libssl1.0-dev

wget 'http://download.oracle.com/berkeley-db/db-4.8.30.NC.tar.gz'
tar -xzvf db-4.8.30.NC.tar.gz
cd db-4.8.30.NC/build_unix/
../dist/configure --enable-cxx --disable-shared --with-pic
make install (as root)

git clone https://github.com/libevent/libevent.git
cd libevent
./autogen.sh && ./configure && make
make install (as root)

cd (your emercoin directory)
./autogen.sh && ./configure CPPFLAGS="-I/usr/local/BerkeleyDB.4.8/include" LDFLAGS="-L/usr/local/BerkeleyDB.4.8/lib"
make
```

**Comment by FndNur1Labs** [comment]
up vote Qt not yet wallet can made but slow connection here. have boostrap/snapshot sir?

**Comment by EvgenijM86** [comment]
No, we don't have those yet.

**Comment by FndNur1Labs** [comment]
hope do ok. we cannot change enviroment fast. for now. we used cli 👍

**Comment by EvgenijM86** [comment]
I have added a patch that have fixed ubuntu 18.04 compilation (due to new version of libssl). It might also work for Debian 9.

**Comment by FndNur1Labs** [comment]
already thank for fixed #debianfans
